### ISSUE TITLE
Thread router device do not reply on Mesh setup Requests 

### ISSUE THREAD

**Issue by canisLupus1313** [openthread/openthread#7068]
1. Git commit id main
1. Network topology:
    - Border Router
    - Joiner Device

Test steps:
Create network with BBR try to attach with joiner. (This step is similar to few test cases in Thread certification eg. RTR-TC-02)

Open thread router doesn't reply on MLE messages eg. DiscoveryRequest, ParentRequest when setting security policy TLV to:
0c040e1013f8 -> NCR=1, CCM=0. This prevents the network to be set up properly.

The reason why MLE messages aren't handled properly is [this](https://github.com/openthread/openthread/blob/main/src/core/thread/mle_router.cpp#L125) check.
There Openthread checks if network can promote non CCM REEDs to Routers, but no check if device is CCM is done.

Is such check available in Open thread or this is missing functionality?

**Comment by jwhui** [comment]
@canisLupus1313 , thanks again for working to address this issue. I did see that you recently submitted #7075.

> And last remark from me IHMO there should be some way to distinguish if the device is CCM or not.

I agree that this is the best path forward to support the Thread Test Harness. Even though a device does not implement CCM, it is useful to indicate whether it should behave as if it were commissioned using CCM. I just submitted #7077 to help support this.

If you agree with the approach in #7077, we will still need to update THCI to apply the appropriate configuration. It would be great if you are willing to help contribute the remaining bits.
